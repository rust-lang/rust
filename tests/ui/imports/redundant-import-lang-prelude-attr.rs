// Check that we detect imports (of built-in attributes) that are redundant due to
// the language prelude and that we emit a reasonable diagnostic.
//~^^ NOTE the item `allow` is already defined by the extern prelude

// Note that we use the term "extern prelude" in the label even though "language prelude"
// would be more correct. However, it's not worth special-casing this.

// See also the discussion in <https://github.com/rust-lang/rust/pull/122954>.

//@ edition: 2018

#![deny(redundant_imports)]
//~^ NOTE the lint level is defined here

use allow; //~ ERROR the item `allow` is imported redundantly

#[allow(unused)]
fn main() {}
