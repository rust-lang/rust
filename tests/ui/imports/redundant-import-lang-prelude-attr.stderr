error: the item `allow` is imported redundantly
  --> $DIR/redundant-import-lang-prelude-attr.rs:15:5
   |
LL | use allow;
   |     ^^^^^ the item `allow` is already defined by the extern prelude
   |
note: the lint level is defined here
  --> $DIR/redundant-import-lang-prelude-attr.rs:12:9
   |
LL | #![deny(redundant_imports)]
   |         ^^^^^^^^^^^^^^^^^

error: aborting due to 1 previous error

